lib/trino-parquet/src/main/java/io/trino/parquet/metadata/ParquetMetadata.java

```diff
@@ -145,6 +143,14 @@ public FileMetaData getParquetMetadata()
         return parquetMetadata;
     }
 
+    private static long getRowGroupStart(List<ColumnChunk> columns, MessageType messageType)
+    {
+        // Note: Do not rely on org.apache.parquet.format.RowGroup.getFile_offset or org.apache.parquet.format.ColumnChunk.getFile_offset
+        // because some versions of parquet-cpp-arrow (and potentially other writers) set it incorrectly
+        ColumnChunkMetadata columnChunkMetadata = toColumnChunkMetadata(columns.getFirst(), null, messageType);
```

Comment:
For my understaning but would this result in reading and processing additional data instead of loading the same from `rowGroup` information ?